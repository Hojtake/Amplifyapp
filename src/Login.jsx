import React from "react";
import ReactDOM from "react-dom";
import FunctionSelection from "./FunctionSelection";
import IkiikiFaceDiagnoseAPI from "./IkiikiFaceDiagnoseAPI";
import classes from "./Login.module.css";

export default class Login extends React.Component{
    constructor(props){
        super(props);
        this.clickLogin = this.clickLogin.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChangePassWord = this.handleChangePassWord.bind(this);
        this.state={message:"",ID:"",password:""}
    }
    
    clickLogin = ()=>{
        
        if(String(this.state.password).match(/[A-Za-z0-9]+/)==null || String(this.state.ID).match(/[A-Za-z0-9]+/) == null){
            
           return; 
        }
        if(this.state.password == "" || this.state.ID == ""){
            
            return;
        }
        const api = new IkiikiFaceDiagnoseAPI();
        api.callLoginAPI(this.state.ID,this.state.password)
        .then(result =>{
            
            //resultがnull,undifinedな場合にエラーとして処理を行う
            if(!result){
                throw new Error();
            }
            if (result.hasLoginAuthenticated) {
                ReactDOM.render(<FunctionSelection ID={result.ID} />, document.getElementById("root"));                
            }else{
                if(!result.message){
                    throw new Error();
                }
                this.setState({message:result.message});
            }
        }).catch(()=>{
            this.setState({message:"予期しないエラーが発生しました。しばらく待ってから再度実行してください。"});
        });
    }

    handleSubmit = (e)=>{
        e.preventDefault();
    }

    handleChangeID = (e)=>{
        this.setState({ID:e.target.value});
    }

    handleChangePassWord = (e)=>{
        this.setState({password:e.target.value});
    }

    render() {  
        return(
            <>
                <h1>イキイキ顔診断　ログイン画面</h1>  
                <div className={classes.main}>
                    <form onSubmit={this.handleSubmit}>
                        ID<br/>
                        <input type="text" className={classes.box1} placeholder="ID" id="ID" pattern="^[0-9a-zA-Z]+$" title="半角英数字のみ入力してください" required value={this.state.ID} onChange={this.handleChangeID}></input><br/>
                        パスワード<br/>
                        <input type="password" className={classes.box2} placeholder="Password" id="password" pattern="^[0-9a-zA-Z]+$" title="半角英数字のみ入力してください"required value={this.state.password} onChange={this.handleChangePassWord}></input>
                        <button type="submit" onClick={this.clickLogin}>ログイン</button>
                        <div className={classes.errMessage} id="errMsg">
                            <p>{this.state.message}</p>
                        </div>
                    </form>
                </div>  
            </>
        )
    }
}