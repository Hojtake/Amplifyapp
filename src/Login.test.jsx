import React from "react";
import {render ,unmountComponentAtNode } from "react-dom";
import { act, renderIntoDocument } from "react-dom/test-utils";
import Login from "./Login";
import TestRenderer from "react-test-renderer"
let container = null;
beforeEach(() =>{
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("API実行後のメッセージが反映されているかのテスト",async ()=>{
    const component = TestRenderer.create(<Login/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})