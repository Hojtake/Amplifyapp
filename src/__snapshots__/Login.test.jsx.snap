// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API実行後のメッセージが反映されているかのテスト 1`] = `
Array [
  <h1>
    イキイキ顔診断　ログイン画面
  </h1>,
  <div
    className="main"
  >
    <form
      onSubmit={[Function]}
    >
      ID
      <br />
      <input
        className="box1"
        id="ID"
        onChange={[Function]}
        pattern="^[0-9a-zA-Z]+$"
        placeholder="ID"
        title="半角英数字のみ入力してください"
        type="text"
        value=""
      />
      <br />
      パスワード
      <br />
      <input
        className="box2"
        id="password"
        onChange={[Function]}
        pattern="^[0-9a-zA-Z]+$"
        placeholder="Password"
        title="半角英数字のみ入力してください"
        type="password"
        value=""
      />
      <button
        onClick={[Function]}
        type="submit"
      >
        ログイン
      </button>
      <div
        className="errMessage"
      >
        <p>
          
        </p>
      </div>
    </form>
  </div>,
]
`;
